server/services/employers.ts

```diff
@@ -168,6 +167,19 @@ export const getSiteByID = async (id: number): Promise<EmployerSite> => {
     throw new Error(`No site found with id ${id}`);
   }
 
+  const currentAllocation = await dbClient.runRawQuery(
+    `
+    SELECT spa.allocation
+      FROM site_phase_allocation spa
+        JOIN (
+          SELECT id FROM phase
+          WHERE CURRENT_DATE BETWEEN start_date AND end_date
+          LIMIT 1
+        ) p ON p.id = spa.phase_id
+      WHERE spa.site_id = $1;
+    `,
+    [site[0].id]
```

Comment:
Verify that `allocationResponse` constains data. If the database doesn't find any valid phase allocations (i.e. there's no allocations for this site at the present time), it will **not** give you a row in the result, causing a server error if you try to read that row. This check should be done right next to the query for readability.

Note: here I am using NaN, but if 0 is preferred, feel free to switch that out.

```suggestion
  const currentAllocation = allocationResponse.length ? allocationResponse[0].allocation : NaN;
```